### ISSUE TITLE
indirect conditional loading

### ISSUE THREAD

**Issue by CarloLucibello** [JuliaLang/Pkg.jl#3348]
Say that my `Plotting` package has a `ContourExt` extension:
```
name = "Plotting"

[weakdeps]
Contour = "d38c429a-6771-53c6-b99e-75d170b6e991"

[extensions]
ContourExt = "Contour"
```
but then in my code I never import `Contour` directly by some package `PkgUsingContour` that depends on `Contour`.
So I wonder if after
```julia
using PkgUsingContour, Plotting
```
the `ContourExt` extension is available. 
In any case, this behavior could be documented more clearly in the docs.

**Comment by KristofferC** [comment]
Yes, no matter how the weak dependency is loaded, the extension will be available.

**Comment by CarloLucibello** [comment]
Very nice, that is the behavior I was hoping for. 
Since the term "loaded package" is used in describing extensions, we could add a corresponding definition in the Glossary section?
